apps/block_scout_web/assets/js/omni.js

```diff
@@ -0,0 +1,163 @@
+  <pre style="font-size: 1vw;display: block;margin-top: 0.5vw;font-weight: bolder;color: white;padding-left: 1vw;">
+  ${JSON.stringify(details, null, 4)}
+  </pre>`;
+  return div;
+};
+
+// expected return {txHash1:true, txHas2:true...}
+const checkTx = async (elements) => {
+  const response = await fetch(url + "/checkOmniTxHashes", {
+    method: "POST",
+    headers: {
+      "Content-Type": "application/json",
+    },
+    body: JSON.stringify({
+      txHashes: elements,
+    }),
+  });
+  if (response.status == 400) {
+    await response.json().then(console.error);
+    return undefined;
+  }
+  return response.json();
+};
+
+const getTxDetails = async (omniTxHash) => {
+  // dummy response
+  const response = await fetch(url + `/omniTx/${omniTxHash}`);
+  const data = await response.json();
+  if (data.length === 0) {
+    return undefined;
+  }
+  return data;
+};
+
+const getAllTxHashElements = () => {
+  return document.querySelectorAll('[data-test="transaction_hash_link"]');
+};
+
+const handletxHashLinks = async () => {
+  const elements = getAllTxHashElements();
+  if (elements.length == 0) {
+    return;
+  }
+  // check local cache
+  const unknownTxList = [];
+  for (let i of elements) {
+    if (i.parentNode.childElementCount <= 2 && txMap[i.innerHTML] === true) {
+      console.log("from cache " + i.innerHTML);
+      i.parentNode.appendChild(customMarker());
+    } else if (txMap[i.innerHTML] === undefined) {
+      unknownTxList.push(i.innerHTML);
+    }
+  }
+  if (unknownTxList.length == 0) {
+    return;
+  }
+  const xChainTxs = await checkTx(unknownTxList);
+  if (xChainTxs === undefined) {
+    return;
+  }
+  for (let i of elements) {
+    if (xChainTxs[i.innerHTML] === true) {
+      txMap[i.innerHTML] = true;
+      console.log("from api " + i.innerHTML);
+      i.parentNode.appendChild(customMarker());
+    } else if (txMap[i.innerHTML] == undefined) {
+      txMap[i.innerHTML] = false;
+    }
+  }
+};
+
+const displayTxDetails = async (txHash) => {
+  const elements = document.querySelectorAll(
+    '[class="card js-ad-dependant-mb-3"]'
+  );
+  if (elements.length != 1) {
+    console.log("could not find unique tx view element");
+    return;
+  }
+
+  const details = await getTxDetails(txHash);
+  if (details === undefined) return;
+
+  // if tx has cross chain data display
+  elements[0].appendChild(customXChainDataView(details));
+};
+
+// sets next repeat only after the function call completes
+const repeatLoop = async (timeMS) => {
+  try {
+    await handletxHashLinks();
+  } catch (err) {
+    console.log(err);
+  } finally {
+    setTimeout(repeatLoop, timeMS);
+  }
+};
+
+const watch = async () => {
+
+  // update api url depending on blockscouts deployed environment
+  if (window.location.hostname.startsWith("testnet")) {
+    url = "testnet-1-xchain.explorer.omni.network";
+  } else if (window.location.hostname.startsWith("staging")) {
+    url = "staging-xchain.explorer.omni.network";
+  }
```

Comment:
These should match the cloudflare domains for xchain indexer api server in each env

Reply:
Nothing is dynamic here all these are static files running on the browser so cannot be parameterized